### PR TITLE
Fix some duplicated branch warnings.

### PR DESCRIPTION
**Pull request by drwells**
I found a few small things to clean up by running `-Wduplicated-cond`.

@jppelteret should there be a different value in 3D?

### CODE REVIEW

**Comment by jppelteret** [comment]
Thanks for picking this up @drwells. No, the heuristically chosen parameter must be the same in 2d and 3d, so its OK for this to be simplified.

Did you mean for the changes to the exception to be added to this PR as well?

**Comment by drwells** [comment]
Yup. The line

```cpp
                  << (arg1 ? "" : "")); 
```

in the exception definition raises this warning.

File: include/deal.II/fe/fe_tools.h [hunk 1]
```diff
@@ -1406,13 +1406,11 @@ namespace FETools
    *
    * @ingroup Exceptions
    */
-  DeclException1(ExcHangingNodesNotAllowed,
-                 int,
-                 << "You are using continuous elements on a grid with "
-                 << "hanging nodes but without providing hanging node "
-                 << "constraints. Use the respective function with "
-                 << "additional AffineConstraints argument(s), instead."
-                 << (arg1 ? "" : ""));
```
Comment (bangerth) [inline include/deal.II/fe/fe_tools.h]: ?? What were we thinking here??

Comment (drwells) [reply include/deal.II/fe/fe_tools.h]: I also found that curious. The original commit (720422590b) used `DeclException1` but did not use the integer argument for anything: 3be5dd5886 added something that used the argument to silence a warning.

Believe it or not, we didn't introduce `DeclExceptionMsg` until 847c44d44a in 2015, so the original author used what was available in 2002.

**Pull request merged**